pyirf/simulations.py

```diff
@@ -168,6 +172,107 @@ def calculate_n_showers_per_energy_and_fov(self, energy_bins, fov_bins):
+            The actual numbers will follow a poissionian distribution around this
+            expected value.
+        """
+        e_fov_offset_integral = self.calculate_n_showers_per_energy_and_fov(
+            energy_bins, fov_offset_bins
+        )
+        viewcone_integral = self.calculate_n_showers_per_fov(
+            [self.viewcone_min, self.viewcone_max] * u.deg
+        )
+        
+        n_bins_pa = len(fov_position_angle_bins) - 1
+        position_angle_integral = np.full(n_bins_pa, viewcone_integral / n_bins_pa)
+        
+        total_integral = e_fov_offset_integral[:, :, np.newaxis] * position_angle_integral
+
+        return total_integral / self.n_showers
+
+    @u.quantity_input(
+        energy_bins=u.TeV, fov_longitude_bins=u.deg, fov_latitude_bins=u.rad
+    )
+    def calculate_n_showers_3d_lonlat(
+        self, energy_bins, fov_longitude_bins, fov_latitude_bins, subpixels=20
+    ):
+        """
+        Calculate number of showers that were simulated in the given
+        energy and 2D fov bins in nominal coordinates.
+
+        This assumes the events were generated uniformly distributed per solid angle,
+        and from a powerlaw in energy like CORSIKA simulates events.
+
+        Parameters
+        ----------
+        energy_bins: astropy.units.Quantity[energy]
+            The energy bin edges for which to calculate the number of simulated showers
+        fov_longitude_bins: astropy.units.Quantity[angle]
+            The FOV longitude bin edges for which to calculate the number of simulated showers
+        fov_latitude_bins: astropy.units.Quantity[angle]
+            The FOV latitude bin edges for which to calculate the number of simulated showers
+
+        Returns
+        -------
+        n_showers: numpy.ndarray(ndim=3)
+            The expected number of events inside each of the
+            ``energy_bins``, ``fov_longitude_bins`` and ``fov_latitude_bins``.
+            Dimension (n_energy_bins, n_fov_longitude_bins, n_fov_latitude_bins)
+            This is a floating point number.
+            The actual numbers will follow a poissionian distribution around this
+            expected value.
+        """
+        
+        fov_mask = _fov_lonlat_grid_overlap_mask(
+            self, fov_longitude_bins, fov_latitude_bins, self.viewcone_max, subpixels=subpixels
+        )
+
+        bin_grid_lon, bin_grid_lat = np.meshgrid(fov_longitude_bins,fov_latitude_bins)
+        bin_area = rectangle_solid_angle(
+            bin_grid_lon[:-1,:-1],
+            bin_grid_lon[1:,1:],
+            bin_grid_lat[:-1,:-1],
+            bin_grid_lat[1:,1:],
+        )
+        viewcone_area = cone_solid_angle(self.viewcone_max) - cone_solid_angle(self.viewcone_min)
+
+        shower_density = self.n_showers / viewcone_area
+
+        fov_integral = shower_density * bin_area
+        e_integral = self.calculate_n_showers_per_energy(energy_bins)
+
+        fov_integral = fov_mask * fov_integral
```

Comment:
Is fov_mask boolean? No, right? It contains a float computed from the subpixel computation, correct?

I wouldn't call it mask then, mask is conventionally used for boolean indexing.